src/Common/TTLCachePolicy.h

```diff
@@ -11,37 +13,47 @@ namespace DB
 class PerUserTTLCachePolicyUserQuota : public ICachePolicyUserQuota
 {
 public:
-    void setQuotaForUser(const String & user_name, size_t max_size_in_bytes, size_t max_entries) override
+    void setQuotaForUser(std::optional<UUID> user_id, size_t max_size_in_bytes, size_t max_entries) override
```

Comment:
`setQuotaForUser()` doesn't look good when its parameter is `std::optional<UUID>`. I mean logically `setQuotaForUser(nullopt, ...)` looks a bit pointless (you set something for an absent user). The function doesn't do anything useful anyway if this parameter is nullopt, so what I suggest is just moving this condition `if (!user_id)` outside of the function and turn the type of the parameter into a plain `const UUID & user_id`. And for other functions `increaseActual()`, `decreaseActual()`, `approveWrite()` it's the same.